onnxruntime/test/python/transformers/test_whisper.py

```diff
@@ -0,0 +1,128 @@
+
+        self.assertEqual(str(optimized_model.model.graph), str(expected_model.model.graph))
+
+    # Attention type #1 in onnx_model_bart.py
+    def test_encoder_attention_fusion_with_skiplayernorm(self):
+        num_heads = 4
+        hidden_size = 64
+        model = create_whisper_encoder_attention(
+            num_heads=num_heads, hidden_size=hidden_size, add_before_layernorm=False
+        )
+        dir = "."
+        model_path = os.path.join(dir, "whisper_encoder_attention_sln.onnx")
+        onnx.save(model, model_path)
+        options = FusionOptions("bart")
+        optimized_model = optimize_model(
+            model_path, model_type="bart", num_heads=num_heads, hidden_size=hidden_size, optimization_options=options
+        )
+        os.remove(model_path)
+        self.verify_fusion(optimized_model, "encoder_attention_with_sln_fused.onnx")
+
+    # Attention type #2 in onnx_model_bart.py
+    def test_decoder_attention_fusion_with_skiplayernorm(self):
+        num_heads = 4
+        hidden_size = 64
+        model = create_whisper_decoder_attention(
+            num_heads=num_heads, hidden_size=hidden_size, add_before_layernorm=False
+        )
+        dir = "."
+        model_path = os.path.join(dir, "whisper_decoder_attention_sln.onnx")
+        onnx.save(model, model_path)
+        options = FusionOptions("bart")
+        optimized_model = optimize_model(
+            model_path, model_type="bart", num_heads=num_heads, hidden_size=hidden_size, optimization_options=options
+        )
+        os.remove(model_path)
+        self.verify_fusion(optimized_model, "decoder_attention_with_sln_fused.onnx")
+
+    # Attention type #4 in onnx_model_bart.py
+    def test_decoder_multihead_attention_fusion(self):
+        num_heads = 4
+        hidden_size = 64
+        model = create_whisper_decoder_multihead_attention(num_heads=num_heads, hidden_size=hidden_size)
+        dir = "."
+        model_path = os.path.join(dir, "whisper_decoder_mha.onnx")
+        onnx.save(model, model_path)
+        options = FusionOptions("bart")
+        options.use_multi_head_attention = True
+        optimized_model = optimize_model(
+            model_path, model_type="bart", num_heads=num_heads, hidden_size=hidden_size, optimization_options=options
+        )
+        os.remove(model_path)
+        self.verify_fusion(optimized_model, "decoder_mha_fused.onnx")
+
+    # Attention type #3 in onnx_model_bart.py
+    def test_decoder_with_past_multihead_self_attention_fusion_with_skiplayernorm(self):
+        num_heads = 4
+        hidden_size = 64
+        model = create_whisper_decoder_with_past_multihead_self_attention(
+            num_heads=num_heads, hidden_size=hidden_size, add_before_layernorm=False
+        )
+        dir = "."
+        model_path = os.path.join(dir, "whisper_decoder_with_past_self_mha.onnx")
+        onnx.save(model, model_path)
+        options = FusionOptions("bart")
```

Comment:
Maybe you can add a model type for "whisper" instead of using "bart".

Reply:
Whisper's model architecture is almost identical to MBART and MBART is a BART variant. From Hugging Face's Whisper implementation of the [encoder](https://github.com/huggingface/transformers/blob/main/src/transformers/models/whisper/modeling_whisper.py#L388) and [decoder](https://github.com/huggingface/transformers/blob/main/src/transformers/models/whisper/modeling_whisper.py#L457):
```
# Copied from transformers.models.mbart.modeling_mbart.MBartEncoderLayer with MBart->Whisper
class WhisperEncoderLayer(nn.Module):
  ...

# Copied from transformers.models.mbart.modeling_mbart.MBartDecoderLayer with MBart->Whisper
class WhisperDecoderLayer(nn.Module):
  ...
```